### PR TITLE
wgengine/netstack: handle 4via6 packets when pinging

### PR DESCRIPTION
**Pull request by andrew-d**
Change-Id: Ib6ebbaa11219fb91b550ed7fc6ede61f83262e89
Signed-off-by: Andrew Dunham <andrew@tailscale.com>

### CODE REVIEW

File: wgengine/netstack/netstack.go [hunk 1]
```diff
@@ -704,6 +708,33 @@ func (ns *Impl) injectInbound(p *packet.Parsed, t *tstun.Wrapper) filter.Respons
...
+	if !viaRange.Contains(destIP) {
+		if tsaddr.IsTailscaleIP(destIP) {
+			return netip.Addr{}, false
+		}
+
+		return destIP, true
+	}
+
+	// This is a 4via6 address; translate the destination to an IPv4
+	// address before forwarding it on. Note that despite ICMP4 and ICMP6
```
Comment (bradfitz) [inline wgengine/netstack/netstack.go:732]: I don't exactly follow this comment but code all looks fine.

File: wgengine/netstack/netstack.go [hunk 2]
```diff
@@ -704,6 +708,33 @@ func (ns *Impl) injectInbound(p *packet.Parsed, t *tstun.Wrapper) filter.Respons
 	return filter.DropSilently
 }
 
+func (ns *Impl) shouldHandlePing(p *packet.Parsed) (netip.Addr, bool) {
```
Comment (bradfitz) [inline wgengine/netstack/netstack.go:711]: conventionally always name "ok bool" results. You can name the Addr `_` if you want.

Comment (bradfitz) [inline wgengine/netstack/netstack.go:711]: docs on this?

**Review by bradfitz — APPROVED** [review]

File: wgengine/netstack/netstack.go (hunk 1 above)
Comment (andrew-d) [reply wgengine/netstack/netstack.go]: Expanded/clarified this a bit.

File: wgengine/netstack/netstack.go [hunk 3]
```diff
@@ -704,6 +708,43 @@ func (ns *Impl) injectInbound(p *packet.Parsed, t *tstun.Wrapper) filter.Respons
 	return filter.DropSilently
 }
 
+// shouldHandlePing returns whether or not netstack should handle an incoming
+// ICMP echo request packet, and the IP address that should be used to make the
```
Comment (bradfitz) [inline wgengine/netstack/netstack.go:712]: the IP address that should be pinged, no? This makes it sound like the source address?

**Review by bradfitz — APPROVED** [review]

**Pull request merged**